Changes to strangeness tables: filter at producer level

ddobrigk opened the pull request:
-> This commit makes the V0Data and the CascData tables fully self-contained and potentially smaller than the base aod::V0 and aod::Cascade tables. In this way, they can also be smaller, and they will always be the tables to analyse by any analysis task - regardless of if they were generated with the finder or the producer. 

-> Also added some comments at the top of all 6 base cxx files for helping people who are trying to find out what's what.

ddobrigk committed:
Changes to strangeness tables: filter at producer level, generic analysis tables

iarsene commented:
Hi @ddobrigk ,
The copyright notice in the cascadeproducer.cxx is not liked by the checkcode test.
Cheers. Ionut

ddobrigk committed:
Add period in copyright notice...

ddobrigk commented:
Hi @iarsene, yes, I noticed now... It's a missing period after "jurisdiction" (in the copyright notice at the very end) that I deleted by accident. Fixed.

ddobrigk commented:
Hi everyone, I am looking at the code complaints and I have to admit I am a bit confused. Asking to add braces to a statement like: 

if ( conditional ) continue; 

Does not really improve code readability IMHO. Are we sure that we want to be this strict regarding O2 commits? Sorry for asking :-)

iarsene commented:
Hi David,
I had a similar question a few weeks ago when this was introduced. This is a convention which was decided upon a long time ago apparently and now it is enforced via these checks. I think some of these strict rules might be changed when the analysis part of O2 will become a separate package, but for now we have to abide by this.
Cheers, Ionut

ddobrigk committed:
Add unnecessary braces
